spinalcordtoolbox/moco.py

```diff
@@ -0,0 +1,894 @@
+    if not os.path.exists(old_mat):
+        os.makedirs(old_mat)
+    # TODO
+    for mat in glob.glob(os.path.join(folder_mat, '*.txt')):
+        sct.copy(mat, old_mat)
+
+    sct.printv('\nloading matrices...', verbose)
+    X = [[[] for i in range(nt)] for i in range(nz)]
+    Y = [[[] for i in range(nt)] for i in range(nz)]
+    X_smooth = [[[] for i in range(nt)] for i in range(nz)]
+    Y_smooth = [[[] for i in range(nt)] for i in range(nz)]
+    for iz in range(nz):
+        for it in range(nt):
+            file =  open(file_mat[it][iz])
+            Matrix = np.loadtxt(file)
+            file.close()
+
+            X[iz][it] = Matrix[0, 3]
+            Y[iz][it] = Matrix[1, 3]
+
+    # Generate motion splines
+    sct.printv('\nGenerate motion splines...', verbose)
+    T = np.arange(nt)
+    if graph:
+        import pylab as pl
+
+    for iz in range(nz):
+
+        spline = scipy.interpolate.UnivariateSpline(T, X[iz][:], w=None, bbox=[None, None], k=3, s=None)
+        X_smooth[iz][:] = spline(T)
+
+        if graph:
+            pl.plot(T, X_smooth[iz][:], label='spline_smoothing')
+            pl.plot(T, X[iz][:], marker='*', linestyle='None', label='original_val')
+            if len(index_b0) != 0:
+                T_b0 = [T[i_b0] for i_b0 in index_b0]
+                X_b0 = [X[iz][i_b0] for i_b0 in index_b0]
+                pl.plot(T_b0, X_b0, marker='D', linestyle='None', color='k', label='b=0')
+            pl.title('X')
+            pl.grid()
+            pl.legend()
+            pl.show()
+
+        spline = scipy.interpolate.UnivariateSpline(T, Y[iz][:], w=None, bbox=[None, None], k=3, s=None)
+        Y_smooth[iz][:] = spline(T)
+
+        if graph:
+            pl.plot(T, Y_smooth[iz][:], label='spline_smoothing')
+            pl.plot(T, Y[iz][:], marker='*', linestyle='None', label='original_val')
+            if len(index_b0) != 0:
+                T_b0 = [T[i_b0] for i_b0 in index_b0]
+                Y_b0 = [Y[iz][i_b0] for i_b0 in index_b0]
+                pl.plot(T_b0, Y_b0, marker='D', linestyle='None', color='k', label='b=0')
+            pl.title('Y')
+            pl.grid()
+            pl.legend()
+            pl.show()
+
+    # Storing the final Matrices
+    sct.printv('\nStoring the final Matrices...', verbose)
+    for iz in range(nz):
+        for it in range(nt):
+            file =  open(file_mat[it][iz])
+            Matrix = np.loadtxt(file)
+            file.close()
+
+            Matrix[0, 3] = X_smooth[iz][it]
+            Matrix[1, 3] = Y_smooth[iz][it]
+
+            file =  open(file_mat[it][iz], 'w')
+            np.savetxt(file_mat[it][iz], Matrix, fmt="%s", delimiter='  ', newline='\n')
+            file.close()
+
+    sct.printv('\n...Done. Patient motion has been smoothed', verbose)
+    sct.printv('------------------------------------------------------------------------------\n', verbose)
+
+
+def combine_matrix(param):
+
+    # required fields
```

Comment:
I think this should become a docstring

Reply:
This function is not used anymore. It was related to Eddy-current distortion correction, which is deprecated (no more maintained). I've removed the related code here: a3952b2490555b522c578e5c53830ba4d1fa40b4 and I have updated this PR body to mention the complete drop of Eddy correction.